apiserver/facades/client/secrets/secrets.go

```diff
@@ -406,16 +407,48 @@ func (s *SecretsAPI) updateSecret(backend provider.SecretsBackend, arg params.Up
 			}
 		}
 	}
-	_, err = s.secretsState.UpdateSecret(uri, fromUpsertParams(arg.UpsertSecretArg))
-	return errors.Trace(err)
+	md, err = s.secretsState.UpdateSecret(uri, fromUpsertParams(arg.AutoPrune, arg.UpsertSecretArg))
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if md.AutoPrune {
+		// If the secret was updated, we need to delete the old unused secret revisions.
```

Comment:
Will the above UpdateSercet() trigger the worker? So the worker will try to do this as well? Given we are using a worker, do we want to do this here - do all the pruning in one place; just let the worker do it

Reply:
Discussed live - this is ok.